pallets/attesters/src/lib.rs

```diff
@@ -321,6 +343,39 @@ pub mod pallet {
             Ok(())
         }
 
+        #[pallet::weight(10_000)]
+        pub fn deregister_attester(origin: OriginFor<T>) -> DispatchResult {
+            let attester = ensure_signed(origin)?;
+
+            // Ensure the attester is registered
+            ensure!(
+                Attesters::<T>::contains_key(&attester),
+                Error::<T>::NotRegistered
+            );
+
+            // Retreive the self-nomination amount
+            let self_nomination =
+                Nominations::<T>::get(&attester, &attester).unwrap_or(Zero::zero());
+
+            // Schedule self-denomination
+            // Calculate the block number when the unnomination can be processed after 2 x shuffling frequency
+            let unlock_block = frame_system::Pallet::<T>::block_number()
+                + T::ShufflingFrequency::get() * 2u32.into();
```

Comment:
Need to define the different phases. Unstaking should be at the beginning of the next next slashable period (+2 from here) so we have the next one to slash, and after the funds can be moved. A slashing period should be a multiple of shuffling or something like that I would say